src/app/components/firebase/firebase.service.js

```diff
@@ -0,0 +1,99 @@
+/* global Firebase:true */
+(function() {
+  'use strict';
+
+  angular
+    .module('gcloudConsole')
+    .factory('firebaseDriver', firebaseDriver);
+
+  /** @nginject */
+  function firebaseDriver(GAuth, GData, $firebaseAuth, $firebaseArray, $firebaseObject) {
+    var rootRef = new Firebase('https://phoenix-storage.firebaseio.com');
+    var userId = GData.getUser().id;
+
+    GAuth.getToken().then(function(tokenObject) {
```

Comment:
Not sure how likely it is to happen, but could this introduce a race condition? Maybe we should make a function for authenticating (with caching) and call that before the various api requests instead of up front like this..?